src/core/shape_inference/include/eye_shape_inference.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/core/validation_util.hpp>
+#include <openvino/opsets/opset9.hpp>
+#include "utils.hpp"
+
+template<class T>
+void shape_infer(const ov::op::v9::Eye* op, const std::vector<T> &input_shapes, std::vector<T> &output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == op->get_input_size() && output_shapes.size() == 1);
+    // output_shape = dims_batch_shape + dims_matrix
+    std::vector<ov::Dimension> dims_batch_shape;
+    std::vector<ov::Dimension> dims_matrix = {ov::Dimension::dynamic(), ov::Dimension::dynamic()};
+    auto& dim_num_rows = dims_matrix[0];
+    auto& dim_num_columns = dims_matrix[1];
+    auto& output_shape = output_shapes[0];
+
+    std::vector<int64_t> num_rows;
+    if (get_data_as_int64<T>(0, op, num_rows, constant_data)) {
+        NODE_VALIDATION_CHECK(op,
+                              num_rows.size() == 1,
+                              "'num_rows' value must be a scalar or 1D tensor. Got: ",
+                              num_rows.size());
+        NODE_VALIDATION_CHECK(op,
+                              num_rows.front() >= 0,
+                              "'num_rows' must be non-negative value. Got: ",
+                              num_rows.front());
+        dim_num_rows = num_rows.front();
+    }
+
+    std::vector<int64_t> num_columns;
+    if (get_data_as_int64<T>(1, op, num_columns, constant_data)) {
+        NODE_VALIDATION_CHECK(op,
+                              num_columns.size() == 1,
+                              "'num_columns' value must be a scalar or 1D tensor. Got: ",
+                              num_columns.size());
+        NODE_VALIDATION_CHECK(op,
+                              num_columns.front() >= 0,
+                              "'num_columns' must be non-negative value. Got: ",
+                              num_columns.front());
+        dim_num_columns = num_columns.front();
+    }
+
+    if (op->get_input_size() == 4) {
+        const auto batch_shape_pshape = input_shapes[3];
+        if (batch_shape_pshape.is_static()) {
+            dims_batch_shape.resize(batch_shape_pshape[0].get_length(), ov::Dimension::dynamic());
+            std::vector<int64_t> batch_shape;
+            if (get_data_as_int64<T>(3, op, batch_shape, constant_data)) {
```

Comment:
Please consider dynamic values too -- look it up [here](https://github.com/openvinotoolkit/openvino/blob/4c659b76697883a409ddbcb4e088a97fa51d7202/src/core/shape_inference/include/broadcast_shape_inference.hpp#L192)